spec.html

```diff
@@ -7680,6 +7680,99 @@ <h1>Jobs and Host Operations to Enqueue Jobs</h1>
       <p>Host environments are not required to treat Jobs uniformly with respect to scheduling. For example, web browsers and Node.js treat Promise-handling Jobs as a higher priority than other work; future features may add Jobs that are not treated at such a high priority.</p>
     </emu-note>
 
+    <emu-clause id="sec-jobcallback-records">
+      <h1>JobCallback Records</h1>
+      <p>The JobCallback is a Record value used to store a callable object and a host-defined ambient context value. Callable objects that are invoked via a Job enqueued by the host may have additional host-defined context. To propagate the state, Job Abstract Closures should not capture and call callable objects directly. Instead, use MakeJobCallback and CallJobCallback.</p>
+      <p>The host hooks HostGetJobCallbackAmbientContext and HostSetJobCallbackAmbientContext control the ambient context.</p>
+      <emu-note>
+        <p>For example, WHATWG HTML uses the ambient context to propagate the incumbent settings object for Promise callbacks.</p>
+      </emu-note>
+      <p>JobCallback Records have the fields listed in <emu-xref href="#table-jobcallback-records"></emu-xref>.</p>
+      <emu-table id="table-jobcallback-records" caption="JobCallback Record Fields">
+        <table>
+          <tbody>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Callback]]
+              </td>
+              <td>
+                A callable object, or *undefined*
```

Comment:
```suggestion
                A callable object, or ~empty~
```
? probably not important either way

Reply:
This should just be a callable object without the possibility of being undefined, as @TimothyGu pointed out.